js/gateio.js

```diff
@@ -2324,19 +2324,19 @@ module.exports = class gateio extends Exchange {
         const entirePath = '/' + type + endPart;
         let url = this.urls['api'][authentication] + entirePath;
         if (authentication === 'public') {
-            if (Object.keys (query).length > 0) {
+            if (Object.keys (query).length) {
```

Comment:
Why would this be a fix?

Reply:
It's due to how the transpiler treats that specific conditional:

- https://github.com/ccxt/ccxt/blob/master/build/transpile.js#L232
- https://github.com/ccxt/ccxt/blob/master/build/transpile.js#L394

Reply:
We have to keep this type of conditional uniform across the lib. In Python and PHP, the same check looks a bit different syntactically, so it's just a special case for checking if the object is empty (has no keys).